Switch go license collector tool + some minor fixes

viktorbenei opened the pull request:
- switch from github.com/godrei/licenses/licenses to github.com/google/go-licenses
- error handling fix in npm analyzer
- added "licence types" summary to the generated output.txt
- added some tests for the golang licence analyzer

viktorbenei committed:
Switch go license collector tool + some minor fixes

- switch from github.com/godrei/licenses/licenses to github.com/google/go-licenses
- error handling fix in npm analyzer
- added "licence types" summary to the generated output.txt
- added some tests for the golang licence analyzer

viktorbenei committed:
npm analyzer fix/improvement

- don't changedir, set the workdir instead
- don't analyze combined output (which includes stderr) only stdout as warnings can be on stderr e.g. {"type":"warning","data":"ghost > request > hawk@3.1.3: This module moved to @hapi/hawk. Please make sure to switch over as this distribution is no longer supported and may contain bugs and critical security issues."}

viktorbenei committed:
npm analyzer fix

Only parse Head and Body in Data if the Type is "table".

Previous version tried to json unmarshall all JSONs to Type+Data,
but in case of "type=warning" the Data is just a string,
so json unmarshal failed, causing a lot of unnecessary warning logs

godrei commented on analyzers/npm/npm.go:
```diff
@@ -62,25 +51,26 @@ func (a *Analyzer) Detect(repoURL, localSourcePath string) (bool, error) {
-		var l npmLicensesList
-		if err := json.Unmarshal([]byte(line), &l); err != nil {
-			log.Warnf("unmarshal yarn licenses output: %s", err)
+		var lType npmLicensesListTypeOnly
```
NOTE: a small comment (or a unit test), similar to what you added for `csvToLicenseInfos` would make it easier to understand the logic here,

viktorbenei replied:
Fixed in commit https://github.com/bitrise-io/github-license-collector/pull/12/commits/83d632c544477213b2aff2b9f8fc8d30614b0ef5

viktorbenei committed:
npm analyzer - unit test for parsing

Similar to the golang parser unit test,
npm analyzer's parser logic was moved into a separate function,
and a unit test was written
to validate the license information parser.

Additionally "logs/" dir was added to gitignore

trapacska reviewed (approved):
LGTM
